helix-term/src/ui/markdown.rs

```diff
@@ -342,9 +342,6 @@ impl Component for Markdown {
 
     fn required_size(&mut self, viewport: (u16, u16)) -> Option<(u16, u16)> {
         let padding = 2;
-        if padding >= viewport.1 || padding >= viewport.0 {
-            return None;
-        }
```

Comment:
This is a footgun right now. Any popup that returns `None` in it's required size implementation leads to a crash.
I wonder if we should just change the signature to return `(u16, u16)` and return `(0, 0) by default. There is no code that actually relies on `required_size` being optional and logically if `required_size` is `None` I would assume that a component requires no size. 

That being said jsut removing this early bail here is not correct as the `viewport.0 - padding` subtractiong would underflow. In my opionion the correct thing to do is change `(viewport.0 - padding)` to `viewport.0.saturation_sub(padding)`. From the `required_size` documentation:

> The returned size might be larger than the viewport if the child is too big to fit.
> In this case the parent can use the values to calculate scroll.